ts/parts/Interaction.ts

```diff
@@ -1264,6 +1259,9 @@ extend(Point.prototype, /** @lends Highcharts.Point.prototype */ {
                 'class': 'highcharts-halo highcharts-color-' +
                     pick(point.colorIndex, series.colorIndex) +
                     (point.className ? ' ' + point.className : ''),
+                'visibility': (
+                    (point.graphic || stateMarkerGraphic) as any
+                ).visibility,
```

Comment:
Would it be better to move this into the condition above? So not rendering halo at all, when point/shared marker is not visible? By the condition above I mean:
https://github.com/highcharts/highcharts/blob/0bb12294b4f4f6c03b4b265488535e0e956b6039/ts/parts/Interaction.ts#L1246

Reply:
Thanks, that is a good point. Will make the necessary change.

It will only be optimal when `(point.graphic || stateMarkerGraphic) as any).visibility` equals `hidden` though. Since `inherit` can be both `'visible'|'hidden'` depending on the parents, it will have to be rendered in this case.

Reply:
@pawelfus Please see latest changes.

Reply:
Looks good, thanks!